portfolio/src/main/java/com/google/sps/data/HighestPriorityFirstScheduler.java

```diff
@@ -0,0 +1,118 @@
+package com.google.sps.data;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+public class HighestPriorityFirstScheduler implements TaskScheduler {
+  private static final Comparator<ScheduledTask> sortByScheduledStartTimeAscending =
+      Comparator.comparing(ScheduledTask::getStartTime);
+
+  /** This method schedules tasks from highest to lowest priority and */
+  public Collection<ScheduledTask> schedule(
+      Collection<CalendarEvent> events,
+      Collection<Task> tasks,
+      Instant workHoursStartTime,
+      Instant workHoursEndTime) {
+
+    List<CalendarEvent> eventsList = new ArrayList<CalendarEvent>(events);
+    List<Task> tasksList = new ArrayList<Task>(tasks);
+
+    TaskQueue taskQueue = new TaskQueue(tasksList, getSchedulingAlgorithmType());
+
+    CalendarEventsGroup calendarEventsGroup =
+        new CalendarEventsGroup(eventsList, workHoursStartTime, workHoursEndTime);
+
+    List<TimeRange> availableTimes = calendarEventsGroup.getFreeTimeRanges();
+    TimeRangeGroup availableTimesGroup = new ArrayListTimeRangeGroup(availableTimes);
+    List<TimeRange> currentAvailableTimes = getAvailableTimeRangesList(availableTimesGroup);
+
+    List<ScheduledTask> scheduledTasks = new ArrayList<ScheduledTask>();
+
+    int availableTimesIndex = 0;
+
+    Instant currentScheduleTime = workHoursStartTime;
+
+    while (availableTimesIndex < currentAvailableTimes.size() && !taskQueue.isEmpty()) {
+      TimeRange availableTimeRange = currentAvailableTimes.get(availableTimesIndex);
+      Task task = taskQueue.peek();
+
+      if (availableTimeRange.start().isAfter(currentScheduleTime)) {
+        currentScheduleTime = availableTimeRange.start();
+      }
+
+      if (!currentScheduleTime
```

Comment:
Consider assigning this value to a variable.

ex:
boolean doesTaskFit = !currentScheduleTime...
if (doesTaskFit) {
  ...

Reply:
Done.